srcStatic/GameModules/ConsoleModule.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "DllModule.h"
+#include <string>
+
+
+class ConsoleModule : public DllModule
+{
+public:
+	ConsoleModule(const std::string& moduleRelativeDirectory);
+
+	void Load() override;
+	bool Unload() override;
+	void Run() override;
+
+	inline std::string Directory() { return moduleDirectory; }
+
+private:
+	typedef void(*LoadModuleFunction)();
+	typedef bool(*UnloadModuleFunction)();
+	typedef void(*RunModuleFunction)();
+
+	std::string moduleDirectory;
+
+	LoadModuleFunction loadModuleFunction = nullptr;
```

Comment:
I wanted to combine these external function hooks with their ini module equivalents and place them in the abstract class DllModule. However, LoadModuleFunction includes an argument in the ini version. The other two, Unload and Run could be combined, but I didn't know if that made since to leave Load orphaned.

I set these = nullptr because the module may not have an associated dll. Otherwise, I'm worried C++ will leave garbage allocated instead of a null value, causing problems when the module's load, run, and unload functions are called.

Reply:
Load differs by a parameter, Unload differs by a return type. The return type isn't really used though, so we could sort of combine things there. Might make sense to log if the return value of Unload indicated an error.

To truly clean this up and unify things, we'd probably need to create a new specification that wasn't backwards compatible.

The `nullptr` initialization looks like a good idea.